Arc servers updatemanagementcenter (#1397)
microsoft/azure_arc: docs/azure_arc_jumpstart/azure_arc_servers/day2/arc_updateManagementCenter/_index.md

---
type: docs
title: "Update Management Center"
linkTitle: "Update Management Center"
weight: 16
description: >
---

## Onboard Azure Arc-enabled servers to Update Management Center

The following Jumpstart scenario will guide you on how to onboard Azure Arc-enabled servers to [Update Management Center](https://learn.microsoft.com/azure/update-center/overview).

> Update management center  is a unified service to help manage and govern updates for all your machines. You can monitor Windows and Linux update compliance across your deployments in Azure, on-premises, and on the other cloud platforms from a single dashboard. Using Update management center, you can make updates in real-time or schedule them within a defined maintenance window.

Reviewer: ```suggestion
Update management center is a unified service to help manage and govern updates for all your machines. You can monitor Windows and Linux update compliance across your deployments in Azure, on-premises, and on the other cloud platforms from a single dashboard. Using Update management center, you can make updates in real-time or schedule them within a defined maintenance window.
```

- **[VMware vSphere Ubuntu VM](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/vmware/vmware_terraform_ubuntu/)**
- **[VMware vSphere Windows Server VM](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/vmware/vmware_terraform_winsrv/)**
- **[Vagrant Ubuntu box](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/vagrant/local_vagrant_ubuntu/)**
- **[Vagrant Windows box](https://azurearcjumpstart.io/azure_arc_jumpstart/azure_arc_servers/vagrant/local_vagrant_windows/)**

Please review the [Update Management Center supported OS documentation](https://learn.microsoft.com/azure/update-center/support-matrix?tabs=azurevm%2Cazurearc-os#supported-operating-systems) and ensure that the VMs you will use for this exercise are supported.

## Prerequisites

- As mentioned, this scenario starts at the point where you already deployed and connected VMs or servers to Azure Arc. In the screenshot below, you can see a Windows and a Linux server that have been connected with Azure Arc and are visible as resources in Azure:

    ![Screenshot Azure Arc-enabled servers](./01.png)

> - [Install or update Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli?view=azure-cli-latest). Azure CLI should be running version 2.14 or later. Use ```az --version``` to check your current installed version.

Reviewer: ```suggestion
- [Install or update Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli?view=azure-cli-latest). Azure CLI should be running version 2.36 or later. Use ```az --version``` to check your current installed version.
```

Reviewer: Just standardizing with the rest of the scenarios

> - To complete this scenario, the user or service principal running the automation need these minimum RBAC permissions: *Resource Policy Contributor* and *Microsoft.Resources/deployments/** on the resource group where you have your Azure Arc-enabled servers.

Reviewer: ```suggestion
- To complete this scenario, the user or service principal running the automation need these minimum Role-based access control (RBAC) permissions: _Resource Policy Contributor_ and _Microsoft.Resources/deployments/_ on the resource group where you have your Azure Arc-enabled servers.
```

- Click the <a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmicrosoft%2Fazure_arc%2Fmain%2Fazure_arc_servers_jumpstart%2FupdateManagementCenter%2Fupdate-management-center-template.json" target="_blank"><img src="https://aka.ms/deploytoazurebutton"/></a> button and enter values for just the ARM template parameters highlighted below. For more information about these parameters, see their description provided in the [*parameters file*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_servers_jumpstart/updatemanagementcenter/update-management-center.parameters.json).

  ![Screenshot showing Azure portal deployment](./02.png)

## Deployment Option 2: ARM template with Azure CLI

As mentioned, this deployment will leverage ARM templates.

- Clone the Azure Arc Jumpstart repository

    ```shell
    git clone https://github.com/microsoft/azure_arc.git
    ```

- Edit the [*parameters file*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_servers_jumpstart/updatemanagementcenter/update-management-center.parameters.json) providing the values that match your configuration as described below:

>     ![Screenshot ARM template parameters file](./03.png)

Reviewer: ```suggestion
    ![Screenshot of ARM template parameters file](./03.png)
```

Reviewer: For accessibility purposes, add "Screenshot of..." before every screenshot

>   > **NOTE: For the *maintenanceTimeZone* parameter, the list of timezones can be obtained by executing *[System.TimeZoneInfo]::GetSystemTimeZones()* in PowerShell. Example: Pacific Standard Time, Romance Standard Time, Middle East Standard Time.**

Reviewer: ```suggestion
  > **NOTE: For the _maintenanceTimeZone_ parameter, the list of time zones can be obtained by executing _[System.TimeZoneInfo]::GetSystemTimeZones()_ in PowerShell. Example: Pacific Standard Time, Romance Standard Time, Middle East Standard Time.**
```

> - Deploy the [*ARM template*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_servers_jumpstart/updatemanagementcenter/update-management-center-template.json) by running the following command, specifying the resource group where you have your Azure Arc-enabled servers:

Reviewer: ```suggestion
- Deploy the [ARM template](https://github.com/microsoft/azure_arc/blob/main/azure_arc_servers_jumpstart/updatemanagementcenter/update-management-center-template.json) by running the following command, specifying the resource group where you have your Azure Arc-enabled servers:
```

    ```shell
    az deployment group create --resource-group <Name of the Azure resource group> \
    --template-file <The update-management-center-template.json template file location> \
    --parameters <The update-management-center.parameters.json template file location>
    ```

- Once the template has completed its run, you should see an output as follows:

    ![Screenshot ARM template execution output](./04.png)

## Update Assessment

> In the resource group that you specified as a parameter, there will be a new policy assignment, *(jumpstart) Configure periodic checking for missing system updates on azure Arc-enabled servers*. This policy, once it's remediated, enables Periodic Assessment. Periodic Assessment is a setting on your machine that enables you to see the latest updates available for your machines and removes the hassle of performing assessment manually every time you need to check the update status. Once you enable this setting, update management center fetches updates on your machine once every 24 hours.

Reviewer: ```suggestion
In the resource group that you specified as a parameter, there will be a new policy assignment named _"(jumpstart) Configure periodic checking for missing system updates on azure Arc-enabled servers_". This policy, once it's remediated, enables Periodic Assessment. 

Periodic Assessment is a setting on your machine that enables you to see the latest updates available for your machines and removes the hassle of performing assessment manually every time you need to check the update status. Once you enable this setting, update management center fetches updates on your machine once every 24 hours.
```

- Go to your **resource group** and click on **Policies**:

    ![Resource Group - Policies](./05.png)

> - You will see the *(jumpstart) Configure periodic checking for missing system updates on azure Arc-enabled servers* policy assigned. Then, click on **Remediation**:
> 
>     ![Policy for checking missing updates](./06.png)
> 
> - Right click on the policy *(jumpstart) Configure periodic checking for missing system updates on azure Arc-enabled servers* policy assigned. Then, click on **Remediate**:

Reviewer: ```suggestion
- You will see the "_(jumpstart) Configure periodic checking for missing system updates on azure Arc-enabled servers_" policy assigned. Then, click on **Remediation**:

    ![Policy for checking missing updates](./06.png)

- Right click on the policy "_(jumpstart) Configure periodic checking for missing system updates on azure Arc-enabled servers_" policy assigned. Then, click on **Remediate**:
```

    ![Policy for checking missing updates - Remediation](./07.png)

- You will see your Azure Arc-enabled server as a resource to be remediated. Click on the **Remediate** button:

    ![Policy for checking missing updates - trigger remediation](./08.png)

- You can see the status of the **Remediation task** and the number of resources remediated:

    ![Policy for checking missing updates - remediation completed](./09.png)

  > **NOTE: You will need to wait some time until the first update assessment is run.**

> - On the update assessment task is run, go back to your **Azure Arc-enabled server** and click on **Extensions**. You will see that a new extension has been installed. Update management center relies on this Azure extension designed to provide all the functionality required to interact with the operating system to manage the assessment and application of updates.

Reviewer: ```suggestion
- On the update assessment task is run, go back to your **Azure Arc-enabled server** and click on **Extensions**, you will see that a new extension has been installed. Update management center relies on this Azure extension designed to provide all the functionality required to interact with the operating system to manage the assessment and application of updates.
```

    ![Update Management Center extension](./10.png)

- Click on **Updates**. Under **Recommended updates**, you will see which are the missing updates on this machine:

    ![Missing updates](./11.png)

- If you click on the **History** tab, you will see the details related to the updates assessment and operations:

    ![Update assessment history](./12.png)

> ## Update Deployment
> 
> On the other hand, as part of this scenario, a new **Maintenance Configuration** resource is created. It allows you to schedule recurring updates deployments on your Azure Arc-enabled servers.

Reviewer: ```suggestion
## Update Deployment

As part of this scenario, a new **Maintenance Configuration** resource is created. It allows you to schedule recurring updates deployments on your Azure Arc-enabled servers.
```

- Go to your **resource group** and click on the new **Maintenance Configuration** resource:

    ![New maintenance configuration resource](./13.png)

> - Review the following **Maintenance Configuration** settings: **Machines**, which includes the Azure Arc-enabled server; **Schedule** of the maintenance configuration resource (it will start one day after you deploy this scenario); and **Updates** settings:

Reviewer: ```suggestion
- Review the following **Maintenance Configuration** settings: **Machines**, which includes the Azure Arc-enabled server; **Schedule** of the maintenance configuration resource (it will start one day after you deploy this scenario), and **Updates** settings:
```

    ![Maintenance configuration resource - machines](./14.png)

    ![Maintenance configuration resource - schedule](./15.png)

    ![Maintenance configuration resource - updates](./16.png)

- Once the **Schedule** triggers the **Maintenance Configuration**, you will be able to see the results. Go back to your **Azure Arc-enabled server**, click on **Updates**, and then on **History**:

    ![Update deployment history](./17.png)

    ![Update deployment history details](./18.png)

## Update Management Center

> From **Update Management Center**, you can oversee the update compliance for your entire fleet of machines in Azure, on-premises, and other cloud environments.

Reviewer: ```suggestion
From the **Update Management Center**, you can oversee the update compliance for your entire fleet of machines in Azure, on-premises, and other cloud environments.
```

> Moreover, you can instantly deploy critical updates to help secure your machines.

Reviewer: ```suggestion
You can also instantly deploy critical updates to help secure your machines.
```

- On the top bar search, type **Update Management Center** and click on it:

    ![Update Management Center](./19.png)

- The **Overview** blade shows you the current updates status of your hybrid environment:

    ![Update Management Center - Overview](./20.png)

- If you click on **Machines**, you will get more details per each individual machine:

    ![Update Management Center - Machines](./21.png)

- Moreover, **History** shows you all the updates activities on your hybrid environment:

    ![Update Management Center - History](./22.png)

- Lastly, under the **Workbooks** section, there is a built-in workbook that provides an unified view for your current updates status:

    ![Update Management Center - Workbooks](./23.png)

## Clean up environment

Complete the following steps to clean up your environment:

- [Delete a maintenance configuration](https://learn.microsoft.com/azure/virtual-machines/maintenance-configurations-cli#delete-a-maintenance-configuration)
- [Remove extension on Azure Arc-enabled server](https://learn.microsoft.com/azure/azure-arc/servers/manage-vm-extensions-cli#remove-extensions)

> - [Delete Azure Policy Assignment](https://learn.microsoft.com/cli/azure/policy/assignment?view=azure-cli-latest#az-policy-assignment-delete)

Reviewer: We should show how to do it so the reader won't have to figure this out. Let's do it "Jumpstart style" :-)
